src/transformers/trainer.py

```diff
@@ -2687,7 +2687,7 @@ def compute_loss(self, model, inputs, return_outputs=False):
 
         if labels is not None:
             if is_peft_available() and isinstance(model, PeftModel):
```

Comment:
```suggestion
            unwrapped_model = unwrap_model(model)
            if is_peft_available() and isinstance(unwrapped_model, PeftModel):
                model_name = unwrapped_model.base_model.model._get_name()
            else:
                model_name = unwrapped_model._get_name()
            if model_name in MODEL_FOR_CAUSAL_LM_MAPPING_NAMES.values():
                loss = self.label_smoother(outputs, labels, shift_labels=True)
            else:
                loss = self.label_smoother(outputs, labels)
            del unwrapped_model
```

Reply:
Don't accept this suggestion otherwise it will create a weird diff but I did this so that you can see what I meant